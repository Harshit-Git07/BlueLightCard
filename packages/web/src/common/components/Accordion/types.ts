export interface AccordionProps {
  title: string;
  children: React.ReactNode;
  onClickOpen?: () => void;
}
