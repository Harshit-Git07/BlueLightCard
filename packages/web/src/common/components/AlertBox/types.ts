export interface AlertBoxProps {
  alertType: string;
  title: string;
  description: any;
}

export interface style {
  textColor: string;
  backgroundColor: string;
}

export interface config {
  [key: string]: style;
}
