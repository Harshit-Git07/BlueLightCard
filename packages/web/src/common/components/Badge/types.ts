export type BgColorString = `bg-${string}`;

export type BadgeProps = {
  label: string;
  color: BgColorString;
  size: 'small' | 'large';
};
