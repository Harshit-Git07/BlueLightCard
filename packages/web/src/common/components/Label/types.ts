export interface LabelProps {
  text: string;
  type: string;
}

export interface style {
  textColor: string;
  backgroundColor: string;
}

export interface config {
  [key: string]: style;
}
