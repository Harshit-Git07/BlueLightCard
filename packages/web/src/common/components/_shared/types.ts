import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import { ChangeEventHandler, ForwardedRef, MouseEventHandler, PropsWithChildren } from 'react';

export type InputFieldWrapperProps = PropsWithChildren & {
  icon?: IconDefinition;
  iconRight?: IconDefinition;
  showRightIcon?: boolean;
  showErrorState?: boolean;
  select?: boolean;
  showSuccessState?: boolean;
  onRightIconClick?: MouseEventHandler;
};

export interface InputFieldSharedProps<E, V = string> {
  name?: string;
  error?: boolean;
  success?: boolean;
  value?: V;
  required?: boolean;
  _ref?: ForwardedRef<unknown>;
  onChange?: ChangeEventHandler<E>;
}
