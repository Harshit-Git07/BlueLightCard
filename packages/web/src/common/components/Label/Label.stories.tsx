import React from 'react';
import { Meta, StoryFn } from '@storybook/react';
import Label from './Label';
import { LabelProps } from './types';

export default {
  title: 'Component System/Label',
  component: Label,
} as Meta;

const Template: StoryFn<LabelProps> = (args) => <Label {...args} />;

export const NormalLabel = Template.bind({});
NormalLabel.args = {
  text: 'Online',
  type: 'normal',
};

export const AlertLabel = Template.bind({});
AlertLabel.args = {
  text: '0/3 codes remaining',
  type: 'alert',
};
