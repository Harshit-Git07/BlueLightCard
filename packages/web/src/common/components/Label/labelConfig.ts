import { config } from './types';

const labelConfig: config = {
  //TODO::Not whitelabel with design tokens revisit
  normal: {
    textColor: 'text-[#202125]',
    backgroundColor: 'bg-[#ECEFF2]',
  },
  alert: {
    textColor: 'text-white',
    backgroundColor: 'bg-shade-system-red-500',
  },
};

export default labelConfig;
