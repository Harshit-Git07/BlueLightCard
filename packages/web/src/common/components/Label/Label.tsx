import { FC } from 'react';
import { LabelProps } from './types';
import labelConfig from './labelConfig';

const Label: FC<LabelProps> = ({ text, type }) => {
  const { textColor, backgroundColor } = labelConfig[type];

  return (
    <div className={`${backgroundColor} inline-flex rounded-full px-3 py-1`}>
      <p className={`${textColor} font-["MuseoSans"] flex items-center text-sm font-semibold`}>
        {text}
      </p>
    </div>
  );
};

export default Label;
