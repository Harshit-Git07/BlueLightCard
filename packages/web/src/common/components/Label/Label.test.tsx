import Label from '@/components/Label/Label';
import { LabelProps } from '@/components/Label/types';
import renderer from 'react-test-renderer';

describe('Label component', () => {
  let props: LabelProps;

  it('should render normal label', () => {
    const props = {
      text: 'Online',
      type: 'normal',
    };
    const component = renderer.create(<Label {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render red label', () => {
    const props = {
      text: '0/3 codes remaining',
      type: 'alert',
    };
    const component = renderer.create(<Label {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
