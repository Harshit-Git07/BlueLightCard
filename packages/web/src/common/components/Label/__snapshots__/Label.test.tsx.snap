// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label component should render normal label 1`] = `
<div
  className="bg-[#ECEFF2] inline-flex rounded-full px-3 py-1"
>
  <p
    className="text-[#202125] font-["MuseoSans"] flex items-center text-sm font-semibold"
  >
    Online
  </p>
</div>
`;

exports[`Label component should render red label 1`] = `
<div
  className="bg-shade-system-red-500 inline-flex rounded-full px-3 py-1"
>
  <p
    className="text-white font-["MuseoSans"] flex items-center text-sm font-semibold"
  >
    0/3 codes remaining
  </p>
</div>
`;
